src/App.tsx

```diff
@@ -1,155 +1,103 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
 
-const USER_ID = 0;
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Notification } from './components/Notification';
+import { SelectTodo } from './types/SelectTodo';
+
+const USER_ID = 10586;
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState<string>('');
+  const [selected, setSelected] = useState<string>('All');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [hideError, setHideError] = useState<boolean>(false);
+
+  useEffect(() => {
+    const timerId = setTimeout(() => {
+      setHideError(false);
+    }, 3000);
+
+    return () => {
+      clearInterval(timerId);
+    };
+  }, []);
+
+  const filterTodo = useCallback(() => {
+    switch (selected) {
+      case SelectTodo.All:
+
+        return todos;
+
+      case SelectTodo.Active:
+        return todos.filter(todo => !todo.completed);
+
+      case SelectTodo.Completed:
+        return todos.filter(todo => todo.completed);
+
+      default:
+        return todos;
+    }
+  }, [selected, todos]);
+
+  const loadTodos = async () => {
+    try {
+      setHideError(false);
+
+      const data = await getTodos(USER_ID);
+
+      if ('Error' in data) {
+        throw new Error('Error - impossible load todo');
+      } else {
+        setTodos(data);
+      }
+    } catch {
+      setHideError(true);
+
+      throw new Error('Error 404 no connection to the server');
+    }
+  };
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const visibleTodos = filterTodo();
+
   return (
     <div className="todoapp">
       <h1 className="todoapp__title">todos</h1>
 
       <div className="todoapp__content">
-        <header className="todoapp__header">
-          {/* this buttons is active only if there are some active todos */}
-          <button type="button" className="todoapp__toggle-all active" />
-
-          {/* Add a todo on form submit */}
-          <form>
-            <input
-              type="text"
-              className="todoapp__new-todo"
-              placeholder="What needs to be done?"
+        <Header
+          query={query}
+          setQuery={setQuery}
+        />
+
+        {todos?.length > 0 && (
```

Comment:
```suggestion
        {!!todos?.length && (
```

Reply:
Add todos is always array, so no need optional chaining